compiler/noirc_evaluator/src/ssa/ssa_gen/context.rs

```diff
@@ -1022,11 +1044,32 @@ fn convert_operator(op: BinaryOpKind) -> BinaryOp {
 impl SharedContext {
     /// Create a new SharedContext for the given monomorphized program.
     pub(super) fn new(program: Program) -> Self {
+        let globals_shared_context = SharedContext::default();
+
+        let globals_id = Program::global_space_id();
+
+        // Queue the function representing the globals space for compilation
+        globals_shared_context.get_or_queue_function(globals_id);
+
+        let mut context = FunctionContext::new(
+            "globals".to_owned(),
+            &vec![],
+            RuntimeType::Brillig(InlineType::default()),
+            &globals_shared_context,
+        );
+        let mut globals = BTreeMap::default();
+        for (id, global) in program.globals.iter() {
+            let values = context.codegen_expression(global).unwrap();
+            globals.insert(*id, values);
+        }
+
         Self {
             functions: Default::default(),
             function_queue: Default::default(),
             function_counter: Default::default(),
             program,
+            globals_context: context.builder.current_function.dfg,
```

Comment:
I'm going to switch to having a pseudo `Function` type to make later code gen simpler, but the logic of the PR will essentially remain the same.

Reply:
Ok I have switched to a `Function`. This PR should be ready for review again @jfecher